Core/src/main/java/com/github/intellectualsites/plotsquared/plot/commands/Buy.java

```diff
@@ -60,7 +66,11 @@ public Buy() {
             if (owner != null) {
                 Captions.PLOT_SOLD.send(owner, plot.getId(), player.getName(), price);
             }
-            plot.removeFlag(PriceFlag.class);
+            PlotFlag<?, ?> plotFlag = GlobalFlagContainer.getInstance().getFlag(PriceFlag.class);
+            PlotFlagRemoveEvent event = PlotSquared.get().getEventDispatcher().callFlagRemove(plotFlag, plot);
+            if(event.getEventResult() != Result.DENY) {
```

Comment:
Space! Also, this should get the flag instance from the plot so that the real value is sent to the event. You can get the plot flag container by using `plot.getFlagContainer()` and then get the flag from there.